scene/3d/navigation_agent_3d.cpp

```diff
@@ -309,6 +309,12 @@ void NavigationAgent3D::update_navigation() {
 		while (o.distance_to(navigation_path[nav_path_index] - Vector3(0, navigation_height_offset, 0)) < target_desired_distance) {
 			nav_path_index += 1;
 			if (nav_path_index == navigation_path.size()) {
+				if (!target_reached) {
+					if (distance_to_target() < target_desired_distance) {
+						emit_signal("target_reached");
+						target_reached = true;
+					}
+				}
```

Comment:
Can you please put this inside a function and call it here and here: https://github.com/godotengine/godot/blob/c10007d32164b17102dfc669bbbf14f87630ed34/scene/3d/navigation_agent_3d.cpp#L109

Reply:
Created a private `_check_distance_to_target` function for this.